packages/features/schedules/lib/use-schedule/useSchedule.ts

```diff
@@ -48,40 +48,46 @@ export const useSchedule = ({
   const searchParams = useSearchParams();
   const routedTeamMemberIds = searchParams ? getRoutedTeamMemberIdsFromSearchParams(searchParams) : null;
   const skipContactOwner = searchParams ? searchParams.get("cal.skipContactOwner") === "true" : false;
-  return trpc.viewer.public.slots.getSchedule.useQuery(
-    {
-      isTeamEvent,
-      usernameList: getUsernameList(username ?? ""),
-      // Prioritize slug over id, since slug is the first value we get available.
-      // If we have a slug, we don't need to fetch the id.
-      // TODO: are queries using eventTypeId faster? Even tho we lost time fetching the id with the slug.
-      ...(eventSlug ? { eventTypeSlug: eventSlug } : { eventTypeId: eventId ?? 0 }),
-      // @TODO: Old code fetched 2 days ago if we were fetching the current month.
-      // Do we want / need to keep that behavior?
-      startTime,
-      // if `prefetchNextMonth` is true, two months are fetched at once.
-      endTime,
-      timeZone: timezone!,
-      duration: duration ? `${duration}` : undefined,
-      rescheduleUid,
-      orgSlug,
-      teamMemberEmail,
-      routedTeamMemberIds,
-      skipContactOwner,
-    },
-    {
-      trpc: {
-        context: {
-          skipBatch: true,
-        },
+
+  const input = {
+    isTeamEvent,
+    usernameList: getUsernameList(username ?? ""),
+    // Prioritize slug over id, since slug is the first value we get available.
+    // If we have a slug, we don't need to fetch the id.
+    // TODO: are queries using eventTypeId faster? Even tho we lost time fetching the id with the slug.
+    ...(eventSlug ? { eventTypeSlug: eventSlug } : { eventTypeId: eventId ?? 0 }),
+    // @TODO: Old code fetched 2 days ago if we were fetching the current month.
+    // Do we want / need to keep that behavior?
+    startTime,
+    // if `prefetchNextMonth` is true, two months are fetched at once.
+    endTime,
+    timeZone: timezone!,
+    duration: duration ? `${duration}` : undefined,
+    rescheduleUid,
+    orgSlug,
+    teamMemberEmail,
+    routedTeamMemberIds,
+    skipContactOwner,
+  };
+
+  const context = {
+    trpc: {
+      context: {
+        skipBatch: true,
       },
-      refetchOnWindowFocus: false,
-      enabled:
-        Boolean(username) &&
-        Boolean(month) &&
-        Boolean(timezone) &&
-        // Should only wait for one or the other, not both.
-        (Boolean(eventSlug) || Boolean(eventId) || eventId === 0),
-    }
-  );
+    },
+    refetchOnWindowFocus: false,
+    enabled:
+      Boolean(username) &&
+      Boolean(month) &&
+      Boolean(timezone) &&
+      // Should only wait for one or the other, not both.
+      (Boolean(eventSlug) || Boolean(eventId) || eventId === 0),
+  };
+
+  if (isTeamEvent) {
+    return trpc.viewer.slots.getTeamSchedule.useQuery(input, context);
```

Comment:
I think conditional hooks are a [React anti-pattern](https://react.dev/reference/rules/rules-of-hooks)

Reply:
That's the use of the hook right? Not what's in it

Reply:
Calling useQuery is technically using a hook. We're wrapping into another hook.

Reply:
Definitely could see that being the case if it returned different data.

Reply:
Checks will tell 🤞

Reply:
> Calling useQuery is technically using a hook. We're wrapping into another hook.

True

Reply:
Checks all passed, @zomars. Is there a refactor you’d like to do here instead?